apps/modernization-ui/src/design-system/inPageNavigation/useInPageNavigation.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react';
+import styles from './InPageNavigation.module.scss';
+
+const useInPageNavigation = () => {
+    useEffect(() => {
+        const sections = Array.from(document.querySelectorAll('section[id]'));
+
+        const scrollHandler = (entries: IntersectionObserverEntry[]) => {
+            entries.forEach((entry) => {
+                const section = entry.target as HTMLElement;
+                const sectionId = section.id;
+                const sectionLink = document.querySelector(`a[href="#${sectionId}"]`);
+                if (entry.intersectionRatio > 0) {
+                    section?.classList.add(styles.visible);
+                    sectionLink?.classList.add(styles.visible);
```

Comment:
suggestion (non-blocking):  I think there is a way to have the class manipulation handled via React only however, that can be for another day.

Reply:
The classNames utility could be used instead here, but the classList is short and easy to keep track of here so I think it's good for now.